### PR TITLE
Remove file watcher related workaround

### PR DESCRIPTION
**Pull request by RomanNikitenko**
Signed-off-by: Roman Nikitenko <rnikiten@redhat.com>

I'm working on CRW-1758.
Within CRW-1758 I tested these changes based on 7.30.x for power related infrastructure.

### What does this PR do?
- There was an issue related to file watchers: the watcher only reported the first directory when creating recursively directories.
 We had to use a workaround to avoid the issue.
  The issue was fixed in the upstream: https://community.theia-ide.org/t/eclipse-theia-v1-12-1-patch-release/1588
 Now we can remove the workaround.
 
 - I have removed the check from `ensureDirExists(filePath: string)` method:
  ```
  if (await fs.pathExists(dirName))
  ``` 
  as it looks like the check is redundant: `fs.mkdirp(dirName)` should not throw an error if a directory already exists.

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/19457

### How to test this PR?

1. Start a workspace from a devfile which contains a debug configuration. 
2. Go to the debug panel and check that the configuration is available for running.

Please repeat it few times, maybe for different workspaces, maybe use restarting a workspace - as the problem was reproduced not permanently. That's why I restarted the happy path tests multiple times - as far as I know that use case was covered by the tests.  

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable

### CODE REVIEW

**Comment by RomanNikitenko** [comment]
[crw-ci-test]

**Comment by RomanNikitenko** [comment]
[crw-ci-test]

**Comment by RomanNikitenko** [comment]
[crw-ci-test]

**Comment by RomanNikitenko** [comment]
[crw-ci-test]

**Comment by RomanNikitenko** [comment]
[crw-ci-test]

**Comment by RomanNikitenko** [comment]
[crw-ci-test]

**Comment by RomanNikitenko** [comment]
[crw-ci-test]

**Review by nickboldt — APPROVED** [review]
If tested successfully against 7.30.x, then I'm +1 for this. I will cherrypick it into 7.30.x so we can get it built into CRW ASAP. 

... I mean, once it gets more approvals from SMEs and maybe more passing tests. :)

**Comment by RomanNikitenko** [comment]
[crw-ci-test]

**Comment by RomanNikitenko** [comment]
[crw-ci-test]

**Comment by RomanNikitenko** [comment]
As it turned out the Happy Path tests reports are broken at the moment.
We can not rely on them - it doesn't matter if it's succeed or failed.

So, I'm waiting on fix for the Happy Path tests and then will restart cycle of testing to get as much reports as possible to confirm that we don't have unstable bug related to file watchers and debug configurations.

**Comment by dmytro-ndp** [comment]
[crw-ci-test --rebuild]

**Comment by dmytro-ndp** [comment]
@RomanNikitenko: please ignore all previous Happy path test results on CRW Jenkins CI: they didn't consume PR changes: https://github.com/eclipse/che/issues/19822

**Comment by dmytro-ndp** [comment]
@RomanNikitenko: https://github.com/eclipse-che/che-theia/pull/1106#issuecomment-843341757 has failed because of CRW Jenkins CI infrastructure problem. So, I will trigger Happy path tests again.

This time there che-theia images, so you can now use [Happy path tests DevFile](https://raw.githubusercontent.com/chepullreq4/pr-check-files/master/che-theia/pr-1106/happy-path-workspace.yaml).

At the same time try-it buttons don't work on DevSandbox because of incompatible changes in devfile:
> Failed to resolve a devfile. Error occurred during creation a workspace from devfile located at `https://raw.githubusercontent.com/chepullreq4/pr-check-files/master/che-theia/pr-1106/simple/che-theia-maven-devfile.yaml`. Cause: Devfile schema validation failed. Error: (/attributes/persistVolumes):The value must be of string type, but actual type is boolean.

**Comment by dmytro-ndp** [comment]
[crw-ci-test]

**Comment by RomanNikitenko** [comment]
[crw-ci-test]

**Comment by RomanNikitenko** [comment]
[crw-ci-test]

**Comment by dmytro-ndp** [comment]
@RomanNikitenko: I have checked it twice to ensure that latest two successful Happy path test executions have been performed on workspace with che-theia image https://quay.io/crw_pr/che-theia:1106 which were built from PR:
- https://github.com/eclipse-che/che-theia/pull/1106#issuecomment-843409747
- https://github.com/eclipse-che/che-theia/pull/1106#issuecomment-843383593
https://main-jenkins-csb-crwqe.apps.ocp4.prod.psi.redhat.com/job/Che/job/e2e/job/minikube/job/pr-check/job/che-theia-pr-check/54/artifact/logs-and-configs/che-config/workspace-pod-description.txt/*view*/
```
theia-iderh1:
    Container ID:   docker://682b5631ff42e898a1a027c14dc53853c1eeba71ac75fd7ffad4ae1e342c14f2
    Image:          quay.io/crw_pr/che-theia:1106
    Image ID:       docker-pullable://quay.io/crw_pr/che-theia@sha256:bdbadbb0faa240a4a118bb09351ce57dbef4da89785e9215925610293045e4c7 
```

**Comment by RomanNikitenko** [comment]
[crw-ci-test]

**Comment by RomanNikitenko** [comment]
[crw-ci-test]

**Comment by RomanNikitenko** [comment]
[crw-ci-test]

**Comment by dmytro-ndp** [comment]
[crw-ci-test]

**Comment by RomanNikitenko** [comment]
[crw-ci-test]

**Comment by RomanNikitenko** [comment]
[crw-ci-test]

**Comment by RomanNikitenko** [comment]
[crw-ci-test]

**Comment by RomanNikitenko** [comment]
I think it's pretty stable, so I'm going to merge it today if there no objections from reviewers.

**Comment by nickboldt** [comment]
> I think it's pretty stable, so I'm going to merge it today if there no objections from reviewers.

Please merge ASAP into 7.30.x branch so we can get it into CRW 2.9

**Comment by RomanNikitenko** [comment]
> > I think it's pretty stable, so I'm going to merge it today if there no objections from reviewers.
> 
> Please merge ASAP into 7.30.x branch so we can get it into CRW 2.9

I really appreciate given approvals (thanks a lot people who already reviewed my PR), but I need at least one approval from code owners to get it merged...

**Review by vzhukovs — APPROVED** [review]

**Review by svor — APPROVED** [review]
LGTM tested on dogfooding instance
screenshot-che-dogfooding apps che-dev x6e0 p1 openshiftapps com-2021 05 19-16_57_08

**Pull request merged**
